File: Assignment39_2.java
// //2. Write a program which accept one number and position from user and off 
// that bit. Return modified number.

import java.util.*;

class Assignment39_2
{
    public static void main(String Args[])
    {
        Scanner sobj = new Scanner(System.in);
        int iRet = 0;

        System.out.println("Enter the Number Here : ");
        int iValue = sobj.nextInt();

        System.out.println("Enter the Position of bit you want to check : ");
        int iPos = sobj.nextInt();


        BitOprations bobj = new BitOprations(iValue,iPos);
        iRet = bobj.BitOff();
        if(iRet == iValue)
        {
            System.out.println("The "+iPos+" Bit was Already Off...Number is : "+iRet);
        }
        else
        {
            System.out.println("Modified Number is : "+iRet);
        }
        
        sobj.close();
    }
}

class BitOprations
{
    public int iNo;
    public int iMask;
    public int iResult;
    public int Pos;

    BitOprations(int A,int B)
    {
        Pos = B;
        iNo = A;
        iMask = 0X00000001;
        iResult = 0;
    }

    int BitOff()
    {
        iMask = iMask << (Pos - 1);
        iResult = iNo & iMask;

        if(iResult == iMask)
        {
            return (iMask ^ iNo);
        }
        else
        {
            return iNo;
        }
    }
}

File: Assignment39_2.cpp
// 2. Write a program which accept one number and position from user and off 
// that bit. Return modified number

#include<iostream>
using namespace std;

class BitOpration
{
    public:
        int iNo;
        int iMask;
        int iResult;
        int Pos;

        BitOpration(int A,int B)
        {
            iNo = A;
            Pos = B;
            iMask = 0X00000001;
            iResult = 0;
        }

        int OffBit()
        {
            iMask = iMask << (Pos - 1);
            iResult = (iNo & iMask);

            if(iResult == iMask)
            {
                return (iNo ^ iMask); 
            }
            else
            {
                return iNo; 
            }

        }

};

int main()
{
    int iValue = 0;
    int iPos = 0;
    int iRet = 0;
    
    cout<<"Enter the Number Here : \n";
    cin>>iValue;

    cout<<"Enter the position of bit that you want to Off : \n";
    cin>>iPos;

    BitOpration bobj(iValue,iPos);
    iRet = bobj.OffBit();

    if(iRet == iValue)
    {
        cout<<"The "<<iPos<<" Bit was Already off..";
        cout<<"Number is : "<<iRet<<"\n";
    }
    else
    {
        cout<<"The Modified Number is : "<<iRet<<"\n";
    }

    return 0;
}

